src/map/pc.c

```diff
@@ -10242,6 +10254,15 @@ void pc_read_skill_tree(void) {
 		clif->skillinfoblock(sd);
 	mapit->free(iter);
 }
+
+int pc_convert_custom_class_name_to_id(const char* name)
+{
+	int val = -1;
+	if (script->get_constant(name, &val))
```

Comment:
This seems a good idea. I wonder if we should use this same method for official jobs, rather than hardcoding that ugly list in pc.c.

I'll take some time to think about this. In case, I'll make a pull request against your fork, to update this.